AmplifyPlugins/Auth/AWSAuthPlugin/Models/SignInResult+Extension.swift

```diff
@@ -8,16 +8,17 @@
 import Amplify
 import AWSMobileClient
 
-extension SignInState {
+extension SignInResult {
 
     func toAmplifyAuthSignInStep() throws -> AuthSignInStep {
-        switch self {
+        switch signInState {
         case .smsMFA:
-            return .confirmSignInWithSMSMFACode
+            let deliveryDetails = AuthCodeDeliveryDetails(destination: .sms(codeDetails?.destination))
+            return .confirmSignInWithSMSMFACode(deliveryDetails, nil)
         case .customChallenge:
-            return .confirmSignInWithCustomChallenge
+            return .confirmSignInWithCustomChallenge(nil)
         case .newPasswordRequired:
-            return .confirmSignInWithNewPassword
+            return .confirmSignInWithNewPassword(nil)
```

Comment:
If `AWSMobileClient` doesn't support this, why are we adding it in our P0?

Reply:
Which one you meant? These are supported by AWSMobileClient